CustomSkillForDataIngestion/QnAIntegrationCustomSkill/Search.cs

```diff
@@ -66,20 +57,28 @@ public static class Search
             QnASearchResultList qnaResponse = null;
             if (data.getAnswer)
             {
-                kbId = await GetKbId(log);
-                qnaRuntimeKey = await GetRuntimeKey(log);
-                runtimeClient = new QnAMakerRuntimeClient(new EndpointKeyServiceClientCredentials(qnaRuntimeKey))
+                
+                try
                 {
-                    RuntimeEndpoint = qnaMakerEndpoint
-                };
+                    kbId = await GetKbId(log);
+                    qnaRuntimeKey = await GetRuntimeKey(log);
+                    runtimeClient = new QnAMakerRuntimeClient(new EndpointKeyServiceClientCredentials(qnaRuntimeKey))
+                    {
+                        RuntimeEndpoint = qnaMakerEndpoint
+                    };
 
-                var qnaOptions = new QueryDTO
+                    var qnaOptions = new QueryDTO
+                    {
+                        Question = data.q,
+                        Top = 1,
+                        ScoreThreshold = 30
+                    };
+                    qnaResponse = await runtimeClient.Runtime.GenerateAnswerAsync(kbId, qnaOptions);
+                }
+                catch (Exception e)
                 {
-                    Question = data.q,
-                    Top = 1,
-                    ScoreThreshold = 30
-                };
-                qnaResponse = await runtimeClient.Runtime.GenerateAnswerAsync(kbId, qnaOptions);
+                    log.LogInformation(e.ToString());
```

Comment:
make this e.message() might give better error message

Reply:
Good catch; will make that update